File: PushyApp/PushyApp/ViewController.m
//
//  ViewController.m
//  PushyApp
//
//  Created by David Reyes Pucheta on 17/11/15.
//  Copyright © 2015 David Reyes Pucheta. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

- (void) requestPermissionToNotify;
- (void) createANotification:(int)secondsInTheFuture;

@end

@implementation ViewController

- (IBAction)scheduleTapped:(id)sender {
    [self requestPermissionToNotify];
    [self createANotification:15];
}

- (void) requestPermissionToNotify{
    
    UIMutableUserNotificationAction *floatAction = [[UIMutableUserNotificationAction alloc] init];
    floatAction.identifier = @"FLOAT_ACTION";
    floatAction.title = @"Float";
    floatAction.activationMode = UIUserNotificationActivationModeBackground;
    floatAction.destructive = YES;
    floatAction.authenticationRequired = NO;
    
    UIMutableUserNotificationAction *stingAction = [[UIMutableUserNotificationAction alloc] init];
    stingAction.identifier = @"STING_ACTION";
    stingAction.title = @"Sting";
    stingAction.activationMode = UIUserNotificationActivationModeForeground;
    stingAction.destructive = NO;
    stingAction.authenticationRequired = NO;
    
    UIMutableUserNotificationCategory *responseCategory = [[UIMutableUserNotificationCategory alloc] init];
    responseCategory.identifier = @"MAIN_CATEGORY";
    [responseCategory setActions:@[stingAction,floatAction] forContext:UIUserNotificationActionContextDefault];
    
    NSSet *categories = [NSSet setWithObjects:responseCategory, nil];
    
    UIUserNotificationType types = UIUserNotificationTypeBadge | UIUserNotificationTypeSound | UIUserNotificationTypeAlert;
    UIUserNotificationSettings *settings = [UIUserNotificationSettings settingsForTypes:types categories:categories];
    [[UIApplication sharedApplication] registerUserNotificationSettings:settings];
}

- (void) createANotification:(int)secondsInTheFuture{
    UILocalNotification *localNotif = [[UILocalNotification alloc] init];
    
    localNotif.fireDate = [[NSDate date] dateByAddingTimeInterval:secondsInTheFuture];
    localNotif.timeZone = nil;
    
    localNotif.alertTitle = @"Alert Title";
    localNotif.alertBody = @"Alert Body";
    localNotif.alertAction = @"okay";
    localNotif.soundName = UILocalNotificationDefaultSoundName;
    
    localNotif.applicationIconBadgeNumber = 123;
    
    localNotif.category = @"MAIN_CATEGORY";
    [[UIApplication sharedApplication] scheduleLocalNotification:localNotif];
}

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: 1 Hour Swift tutorial/1 Hour Swift tutorial/ViewController.swift
//
//  ViewController.swift
//  1 Hour Swift tutorial
//
//  Created by David Reyes Pucheta on 25/09/15.
//  Copyright © 2015 David Reyes Pucheta. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var intNumber : Int = 0
    var doubleDecimal : Double = 0.23
    var doubleFloa : Float = -233.3
    var stringName : String = "David Reyes"
    var isBoolean : Bool = false

    @IBOutlet weak var lbl: UILabel!
    @IBOutlet weak var btnButton: UIButton!
    @IBOutlet weak var txtNumber: UITextField!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    @IBAction func btnButtonAction(sender: UIButton) {
        intNumber = Int(txtNumber.text!)!
        changeText()
    }
    
    func changeText (){
        lbl.text = "\(intNumber)"
        btnButton.setTitle("\(intNumber)", forState: UIControlState.Normal)
    }
    
    func changeTextAtStart (){
        lbl.text = "Start"
        btnButton.setTitle("Start text", forState: UIControlState.Normal)
    }

}

